### ISSUE TITLE
Remove deprecated t.p.t.ThreadPool.dispatch/dispatchWithCallback

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5037]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5037|
|Type|task|
|Created|2011-04-07 17:23:09Z|
|Branch|https://github.com/twisted/twisted/tree/threadpool-dispatch-5037|

[twisted/python/threadpool.py](twisted/python/threadpool.py)'s `ThreadPool` class has `dispatch` and `dispatchWithCallback` methods that were deprecated in twisted 8.2, so they should be removed now.

<details><summary>Searchable metadata</summary>

```
trac-id__5037 5037
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_threadpool_dispatch_5037 branches/threadpool-dispatch-5037
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1302196989000000 1302196989000000
changetime__1328548531000000 1328548531000000
version__None None
owner__thijs thijs
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [33507]) Branching to 'threadpool-dispatch-5037'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [33508]) removed the deprecated ThreadPool.dispatch and dispatchWithCallback, refs [#5037](https://github.com/twisted/twisted/issues/5037)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess set owner to @thijstriemstra|
|-|-|

ReviewProcess#Newsfiles requests that news files for removals include when the removed code was deprecated. Other than that, [build results look good](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/threadpool-dispatch-5037|), please merge!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [33515]) address review comments, refs [#5037](https://github.com/twisted/twisted/issues/5037)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [33516]) Merge threadpool-dispatch-5037: Removed the deprecated `dispatch` and `dispatchWithCallback` methods from t.p.threadpool.ThreadPool (deprecated since 8.0)

Author: thijs
Reviewer: jesstess
Fixes: [#5037](https://github.com/twisted/twisted/issues/5037)
